### PR TITLE
Jv-pecs task solution

### PR DESCRIPTION
**Pull request by mkazarian-1**

### CODE REVIEW

File: src/main/java/core/mate/academy/model/Bulldozer.java [hunk 1]
```diff
@@ -5,11 +5,32 @@
...
+    }
+
+    public int getWeight() {
+        return weight;
+    }
+
+    public Bulldozer setWeight(int weight) {
+        this.weight = weight;
+        return this;
+    }
```
Comment (koval-kn) [inline src/main/java/core/mate/academy/model/Bulldozer.java]: Do the same for Excavator and Truck classes
1. use constructor-based fields initialization
2. setters should have `void` return type
```suggestion
    public Bulldozer(int enginePower, int weight) {
    this.enginePower = enginePower;
    this.weight = weight;
    }

    @Override
    public void doWork() {
        System.out.println("Bulldozer started to work");
    }

    public int getEnginePower() {
        return enginePower;
    }

    public int getWeight() {
        return weight;
    }
```

File: src/main/java/core/mate/academy/service/MachineProducer.java [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
 
-public interface MachineProducer {
+public interface MachineProducer<T extends Machine> {
     /**
      * In your implementations - create 2-3 models, add them to the list and return
+     *
      * @return - the list of models
      */
-    List<Object> get();
+    List<? extends T> get();
```
Comment (koval-kn) [inline src/main/java/core/mate/academy/service/MachineProducer.java]:
```suggestion
    List<T> get();
```

File: src/main/java/core/mate/academy/service/MachineService.java [hunk 3]
```diff
@@ -16,7 +16,7 @@ public interface MachineService {
      * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
      * @return the list of machines
      */
-    List<Object> getAll(Class type);
+    List<Machine> getAll(Class<? extends T> type);
```
Comment (koval-kn) [inline src/main/java/core/mate/academy/service/MachineService.java]:
```suggestion
    List<T> getAll(Class<? extends T> type);
```

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 4]
```diff
@@ -1,7 +1,46 @@
...
+
+        if (Bulldozer.class.equals(type)) {
+            machines = new BulldozerProducer().get();
+        } else if (Truck.class.equals(type)) {
+            machines = new TruckProducer().get();
+        } else if (Excavator.class.equals(type)) {
+            machines = new ExcavatorProducer().get();
+        }
+
+        return new ArrayList<>(machines);
```
Comment (koval-kn) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java]: producers should be `private final` class-level fields
```suggestion
        if (Bulldozer.class.equals(type)) {
            return bulldozerProducer().get();
        } else if (Truck.class.equals(type)) {
            return truckProducer().get();
        } else if (Excavator.class.equals(type)) {
            return excavatorProducer().get();
        }
```

**Review by koval-kn — CHANGES_REQUESTED** [review]

**Review by sokimaaa — APPROVED** [review]

**Pull request closed without merging**